include/boost/gil/rasterization/ellipse.hpp

```diff
@@ -0,0 +1,181 @@
+//
+// Copyright 2021 Prathamesh Tagore <prathameshtagore@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+#ifndef BOOST_GIL_RASTERIZATION_ELLIPSE_HPP
+#define BOOST_GIL_RASTERIZATION_ELLIPSE_HPP
+
+#include <array>
+#include <vector>
+#include <iostream>
+
+namespace boost { namespace gil {
+
+/// \defgroup EllipseRasterization
+/// \ingroup Rasterization
+/// \brief Ellipse rasterization algorithms.
+
+/// \ingroup EllipseRasterization
+/// \brief Performs ellipse rasterization using midpoint algorithm.
+struct midpoint_elliptical_rasterizer
+{
+    /// \brief Returns a vector containing co-ordinates of first quadrant points which lie on
+    /// rasterizer trajectory of the ellipse.
+    /// \param semi_axes - Array containing half of lengths of horizontal and vertical axis respectively.
+    std::vector<std::array<std::ptrdiff_t, 2>> obtain_trajectory(
+                                                    const std::array<unsigned int, 2> semi_axes)
+    {
+        // Citation : J. Van Aken, "An Efficient Ellipse-Drawing Algorithm" in IEEE Computer Graphics and Applications, vol. 4, no. 09, pp. 24-35, 1984.
```

Comment:
Does this citation mean your algorithm is based on this paper?

Reply:
Yes, algorithm for obtaining trajectory points used inside ```obtain_trajectory()``` is based on the mentioned paper.